### PR TITLE
[M1080] citation handling

### PR DESCRIPTION
**Pull request by saanobhaai**
## Summary by CodeRabbit

- **New Features**
  - Added `user_citation` field to the project model for optional text input.
  - Introduced `suggested_citation` field across multiple models to allow for optional citation information.
  - Implemented a new `citation_name` property in the Profile class for formatted user citations.
  - Enhanced serializers to include `suggested_citation` for improved citation handling in outputs.

- **Bug Fixes**
  - Improved error handling in various update methods to ensure better validation and reporting.

- **Tests**
  - Added unit tests for citation-related functionalities to validate expected behaviors.

- **Documentation**
  - Updated documentation to reflect new fields and functionalities related to citations.

### CODE REVIEW

**Comment by saanobhaai** [comment]
@coderabbit review

**Comment by saanobhaai** [comment]
[citation API functionality](https://trello.com/c/XLADRS8M/1080-citation-api-functionality)

File: src/api/resources/project.py [hunk 1]
```diff
@@ -58,19 +62,43 @@
...
         exclude = []
+        hidden_fields = ["default_citation", "user_citation", "citation_retrieved_text"]
         additional_fields = ["countries", "num_sites"]
 
+    def _get_profiles(self, obj):
+        if self._cached_profiles is None:
+            self._cached_profiles = get_profiles(obj)
+        return self._cached_profiles
+
+    def get_citation_retrieved_text(self, obj):
```
Comment (gridcell) [inline src/api/resources/project.py:91]: Should change or alias the `get_citation_retrieved_text` function in the import. I see this in other files as well

File: src/api/resources/project.py [hunk 2]
```diff
@@ -101,7 +130,7 @@ def get_num_sample_units(self, obj):
         return num_sample_units
 
 
-class ProjectSerializer(DynamicFieldsMixin, BaseProjectSerializer):
```
Comment (gridcell) [inline src/api/resources/project.py:104]: is DynamicFieldsMixin not needed anymore?

Comment (saanobhaai) [reply src/api/resources/project.py:104]: @gridcell I moved it to `BaseProjectSerializer` (i.e. up in the inheritance chain). Does that work?

**Review by gridcell — APPROVED** [review]

**Pull request merged**